### DISCUSSION TITLE
Upload file size restrictions

### DISCUSSION THREAD

**Discussion by TheOtherBrian1** [supabase/supabase#27431] [Troubleshooting]
_This is a copy of a troubleshooting article on Supabase's docs site. It may be missing some details from the original. View the [original article](https://supabase.com/docs/guides/troubleshooting/upload-file-size-restrictions-Y4wQLT)._

You can view the maximum file upload size from the [docs](https://supabase.com/docs/guides/storage/uploads/file-limits). It differs based on your plan and on any bucket or global limits you might have already set.

## Edit the maximum file upload size

There are two ways to edit the maximum file upload size:

*   Global file upload size limit
*   Bucket-level file upload size limit

### Global file upload size limit

To edit the global file upload size limit, change the value of *Global file size limit* in the top section of [Storage Settings](https://supabase.com/dashboard/project/_/storage/settings).

If they are set higher than your new global limit, you’ll be asked to edit or remove any existing bucket-level limits first. The global limit takes precedence.

### Bucket-level file upload size limit

Follow these steps to edit the maximum file upload size at the bucket level:

1.  Open a bucket in the [Storage Buckets](https://supabase.com/dashboard/project/_/storage/buckets/) page.
2.  Select the overflow menu (three vertical dots icon), and then select *Edit bucket*.
3.  Enable *Restrict file size* and then set the file size limit for that bucket.

You won’t be able to set a limit that is greater than the global file upload size limit, as that takes precedence.

## File size restrictions

Upload methods impose differing file size restrictions:

*   [Standard uploads can only transfer up to 5 GBs](https://supabase.com/docs/guides/storage/uploads/standard-uploads?queryGroups=language\&language=js). However, for files above 6MB, the below methods are more performant and reliable.
*   [Resumable](https://supabase.com/docs/guides/storage/uploads/resumable-uploads) and [S3](https://supabase.com/docs/guides/storage/uploads/resumable-uploads) uploads can support transfers up to 50GB in size.

**Comment by GaryAustin1** [comment]
Adding to this.
A common issue is to upgrade the main storage size setting, but not know to adjust the bucket setting also.
You would likely see this error: ` 'Payload too large', 'message': 'The object exceeded the maximum allowed size'`
